# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../fourier/initialize-padding.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- fourier-0.3.0/fourier.c.vanilla	2007-12-01 12:07:25.000000000 +0100
+++ fourier-0.3.0/fourier.c	2007-12-01 12:09:34.000000000 +0100
@@ -319,11 +319,17 @@
     }
     else
     {
+
+      // zero padding
+      for (col = 0; col < w; ++col)
+        for (row = h; row < h+padding; ++row)
+          fft_real[col*(h+padding)+row] = .0;   
+
       for (row = 0; row < h; row++)
       {
         // Restore row order
         row2 = restore_row (row, h);
-        
+       
         for(col = 0; col < w; col++)
         {
           col2 = restore_col (col, w);
